src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1809,6 +1811,29 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
             }
         }
 
+        /// <summary>
+        /// Filters out packages that are considered invalid by Helix. This includes any package
+        /// that has a coordinate with a special value like NaN or Infinite.
+        /// </summary>
+        /// <param name="packages">Original packages to render</param>
+        /// <returns>List of packages considered valid</returns>
+        private IEnumerable<HelixRenderPackage> FilterOutInvalidPackages(IEnumerable<HelixRenderPackage> packages)
+        {
+            List<HelixRenderPackage> result = new List<HelixRenderPackage>();
+
+            foreach (var package in packages)
+            {
+                if (!package.Points.Positions.Any(v => v.IsInvalid()) &&
+                    !package.Lines.Positions.Any(v => v.IsInvalid()) &&
+                    !package.Mesh.Positions.Any(v => v.IsInvalid()))
+                {
```

Comment:
I wonder if this will cause a significant slowdown.

Reply:
it sure seems like it could - for an alternative, what about checking this during package creation?

Reply:
The checks should be quick. The amount will grow with geometry, but should be a very low percentage in comparison.

@mjkkirschner What do you mean by package creation? To check it as soon as the geometry is added to the package?

Reply:
yes, exactly, since we already need to iterate over each vertex at that point. It may not be possible though if some render packages are created in LibG code.

Reply:
`GetRenderPackagesFromMirrorData` is the function I am thinking of.

Reply:
But - it's probably better to profile this than ask you to change your implementation without reason.

Reply:
Sounds good 👍